### PR TITLE
CP-48057: Add unit tests for drivers/scheduler.c

### PR DESCRIPTION
**Pull request by KevinLCtx**
New unit tests for `drivers/scheduler.c`.

Koji job: 657827

### CODE REVIEW

File: mockatests/drivers/test-scheduler.c [hunk 1]
```diff
@@ -0,0 +1,1315 @@
...
+  scheduler_for_each_event(s, event){
+    len += 1;
+  }
+  return len;
+}
+
+void fake_event_cb (event_id_t id, char mode, void *private) {}
+
+void
+test_scheduler_set_max_timeout(void **state)
```
Comment (MarkSymsCtx) [inline mockatests/drivers/test-scheduler.c:82]: `test_scheduler_set_max_timeout_negative` ?

File: mockatests/drivers/test-scheduler.c [hunk 2]
```diff
@@ -0,0 +1,1315 @@
...
+  assert_ptr_equal(e->cb, &fake_event_cb);
+  assert_ptr_equal(e->private, &private);
+  assert_ptr_equal(e->masked, 0);
+
+  assert_int_equal(e->deadline.tv_sec,  fake_gettimeofday.tv_sec + timeout.tv_sec);
+  assert_int_equal(e->deadline.tv_usec, fake_gettimeofday.tv_usec + timeout.tv_usec);
+}
+
+void
+test_scheduler_set_timeout_inf(void **state)
```
Comment (MarkSymsCtx) [inline mockatests/drivers/test-scheduler.c:204]: `test_scheduler_register_event_timeout_inf` ?

File: mockatests/drivers/test-scheduler.c [hunk 3]
```diff
@@ -0,0 +1,1315 @@
...
+  fake_gettimeofday = (struct timeval){ .tv_sec = 0, .tv_usec = 0};
+  (void)scheduler_register_event(&s, md, fd, to, &fake_event_cb, NULL);
+
+  scheduler_prepare_events(&s);
+
+  assert_int_equal(s.max_fd, -1);
+}
+
+void
+test_scheduler_with_no_eventS_will_timeout(void **state)
```
Comment (MarkSymsCtx) [inline mockatests/drivers/test-scheduler.c:1076]: looks like we have a capital S on the end of eventS ?

File: mockatests/drivers/test-scheduler.c (hunk 1 above)
Comment (KevinLCtx) [reply mockatests/drivers/test-scheduler.c:82]: Added in b5a2f60.

File: mockatests/drivers/test-scheduler.c (hunk 2 above)
Comment (KevinLCtx) [reply mockatests/drivers/test-scheduler.c:226]: Added in b5a2f60.

File: mockatests/drivers/test-scheduler.c (hunk 3 above)
Comment (KevinLCtx) [reply mockatests/drivers/test-scheduler.c]: Fixed in b5a2f60.

**Review by MarkSymsCtx — APPROVED** [review]

**Review by rdn32 — APPROVED** [review]

**Pull request merged**